cognite/extractorutils/unstable/configuration/models.py

```diff
@@ -139,23 +147,40 @@ def __repr__(self) -> str:
         return self._expression
 
 
-class _ConnectionParameters(ConfigModel):
-    gzip_compression: bool = False
-    status_forcelist: list[int] = Field(default_factory=lambda: [429, 502, 503, 504])
-    max_retries: int = 10
-    max_retries_connect: int = 3
-    max_retry_backoff: TimeIntervalConfig = Field(default_factory=lambda: TimeIntervalConfig("30s"))
-    max_connection_pool_size: int = 50
-    ssl_verify: bool = True
-    proxies: dict[str, str] = Field(default_factory=dict)
+class RetriesConfig(ConfigModel):
+    max_retries: int = Field(default=10, ge=-1)
+    max_backoff: TimeIntervalConfig = Field(default_factory=lambda: TimeIntervalConfig("30s"))
     timeout: TimeIntervalConfig = Field(default_factory=lambda: TimeIntervalConfig("30s"))
 
 
+class SslCertificatesConfig(ConfigModel):
+    verify: bool = True
+    allow_list: list[str] | None = None
+
+    @field_validator("allow_list", mode="before", json_schema_input_type=str | list[str] | None)
+    @classmethod
+    def cast_thumbprints(cls, thumbprints: str | list[str] | None) -> list[str] | None:
+        if thumbprints is None:
+            return None
+        if isinstance(thumbprints, str):
+            return [scope.strip() for scope in thumbprints.split(",")]
+        return thumbprints
+
+
+class _ConnectionParameters(ConfigModel):
```

Comment:
I realize this is the same name as before, but I'm not sure this type should actually be private? It makes it impossible to safely construct a `ConnectionConfig` for users of the utils. What was the reasoning for this @mathialo?